src/effects/VST3/VST3Effect.cpp

```diff
@@ -339,6 +339,68 @@ bool VST3Effect::SupportsAutomation() const
    return false;
 }
 
+
+bool VST3Effect::FetchSettings(VST3EffectSettings& settings) const
+{
+   // partly copied from ::SaveSettings + some adaptations made
+
+   if (mEditController == nullptr)
+      return false;
+
+   using namespace Steinberg;
+
+   auto theCount = mEditController->getParameterCount();
+
+   for (int i = 0, count = mEditController->getParameterCount(); i < count; ++i)
+   {
+      Vst::ParameterInfo parameterInfo{ };
+      if (mEditController->getParameterInfo(i, parameterInfo) == kResultOk)
+      {
+         if (parameterInfo.flags & Vst::ParameterInfo::kCanAutomate)
+         {
+
+            Steinberg::Vst::ParamID    key = parameterInfo.id;
+            Steinberg::Vst::ParamValue value = mEditController->getParamNormalized(parameterInfo.id);
+
+            settings.mValues[key] = value;
```

Comment:
Is that possible to store pointer to the controller instead?

Reply:
Yes, instead of storing parameter values in `settings` use `settings` object to access underlying `EditController`

Reply:
I'm missing some context.  These lines are no longer in the last commit of this branch.

Are you suggesting that a pointer to `EditController` should be carried around in the `EffectSettings` object?  I don't like that.  `EffectSettings` should be thought of as a "frozen" description of instance state, not using any third-party library handles.  Thus our fetching and storing functions to synchronize frozen with live state, one way or the other.

The `EffectSettings` is something we can pass across inter-thread boundaries, and also copy from one live instance to the other by means of fetching from one and storing to the other.

Maybe it is not the most efficient way for every effect protocol, but it seems to me to have the generality that we need that would work with all of them in one unifying framework.